makefiles/release.mk

```diff
@@ -0,0 +1,139 @@
+# Makefile for Path project
+# Supports building, testing, releasing, and CI workflow testing
+
+#############################################
+##          Configuration variables        ##
+#############################################
+VERSION ?= $(shell git describe --tags --always)
+DATE ?= $(shell date -u +%Y-%m-%dT%H:%M:%SZ)
+BINARY_NAME ?= path
+# Supported build platforms
+PLATFORMS ?= linux/amd64 linux/arm64 darwin/amd64 darwin/arm64
+# Build flags
+LDFLAGS := -ldflags="-X main.Version=$(VERSION) -X main.Date=$(DATE)"
+# Path to GitHub Actions workflows
+GH_WORKFLOWS := .github/workflows
+# Output directories
+RELEASE_DIR := release
+BIN_DIR := bin
+# Architecture detection for M-series Macs
+ARCH := $(shell uname -m)
+ifeq ($(ARCH),arm64)
+  # Check if running on macOS
+  ifeq ($(shell uname),Darwin)
+    ACT_ARCH_FLAG := --container-architecture linux/amd64
+  endif
+endif
+
+#############################################
+##             Build commands              ##
+#############################################
+
+.PHONY: path_build
+path_build: ## Build the binary for local development
+	@mkdir -p $(BIN_DIR)
+	go build $(LDFLAGS) -o $(BIN_DIR)/$(BINARY_NAME) ./cmd
+
+.PHONY: path_release
+path_release: ## Build release binaries for all supported platforms
+	@mkdir -p $(RELEASE_DIR)
+	@for platform in $(PLATFORMS); do \
+		os=$$(echo $$platform | cut -d/ -f1); \
+		arch=$$(echo $$platform | cut -d/ -f2); \
+		echo "Building for $$os/$$arch..."; \
+		output_name="$(BINARY_NAME)_$(VERSION)_$${os}_$${arch}"; \
+		GOOS=$$os GOARCH=$$arch go build $(LDFLAGS) -o $(RELEASE_DIR)/$$output_name ./cmd; \
+		tar -czf $(RELEASE_DIR)/$$output_name.tar.gz -C $(RELEASE_DIR) $$output_name; \
+		rm $(RELEASE_DIR)/$$output_name; \
+	done
+	@echo "✅ Release binaries built in the '$(RELEASE_DIR)' directory"
+
+#############################################
+##          Versioning commands            ##
+#############################################
+
+.PHONY: path_release_tag
+path_release_tag: ## Tag a new release with specified type (bug, minor, major)
```

Comment:
#PUC with more detail on version tags, ie. an example of an actual tag for each type.